### PR TITLE
feat: support fn in headers option

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
yes

### Motivation / Use-Case
allow headers function for customizability

### Breaking Changes
no

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]

**Review by alexander-akait — COMMENTED** [review]
Please add small test

**Comment by anshumanv** [comment]
sorry for delay, will do today

**Comment by anshumanv** [comment]
@alexander-akait done

File: test/server/headers-option.test.js [hunk 1]
```diff
@@ -55,6 +55,32 @@ describe('headers option', () => {
...
+      );
+      req = request(server.app);
+    });
+
+    afterAll(testServer.close);
+
+    it('GET request with headers as a function', (done) => {
+      // https://github.com/webpack/webpack-dev-server/pull/1650#discussion_r254217027
+      const expected = ['v7', 'v8', 'v9'].includes(
+        process.version.split('.')[0]
```
Comment (alexander-akait) [inline test/server/headers-option.test.js:78]: We don't support these Node.js versions https://nodejs.org/api/process.html#process_process_version :smile:

Comment (anshumanv) [reply test/server/headers-option.test.js]: done

**Pull request merged**